### ISSUE TITLE
Support for emacs 29's built-in `treesit` library

### ISSUE THREAD

**Issue by rweir** [emacs-vs/ts-docstr#13]
Hi!

As you probably know, emacs29 includes built in support for tree-sitter grammars with a library called `treesit`.  Unfortunately it's incompatible with the older out-of-tree `tree-sitter.el` that this package is based on.  Are you interested in porting it / would you accept a patch to port it, which would make this package only work on emacs 29+?

**Comment by jcs090218** [comment]
Yes, I would like to support the built-in `treesit`. However, I am still using the external package `tree-sitter.el`, so it will take a while. But I do accept PRs as long as this package still supports the `tree-sitter.el`. Therefore, the best approach is to support them both in this package.

**Comment by ypytomets** [comment]
Thank you for this plugin.  Any updates according to the migration on `treesit`?  Since both `docstr` and `ts-docstr` are not working with the modern treesit-enabled modes, like `scala-ts-mode`.  And yet it looks like it's the only library that able to highlight documentation comments like `Scaladoc` (partially, unfortunately there's no fancy markdown stuff). 🙏 @jcs090218
